### PR TITLE
Important virus fixes

### LINKED ISSUE #42975: Beard virus not beardifying
Round ID: 102868
Local commit: ace9eeb1d4fb2c1104f417a87acf19f2c20cbf33

Made a Facial Hypertrichosis virus and it didn't do anything besides spam "You feel manly." It's supposed to make beards grow.

### LINKED ISSUE #42976: Necrotic Metabolism virus is not very good
Local commit: ace9eeb

Corpses infected with Necrotic Metabolism and on-contact transmitabillity weren't infecting others. It makes the symptom rather pointless unless the whole station is skeletons.

Also, the stats of the symptom make it unviable even if it worked correctly. It really ought to have a positive transmission value such as 2.

### PR DESCRIPTION
**Pull request by pubby**
_**Notice anything wrong here?**_

	if(A.spread_flags &= DISEASE_SPREAD_AIRBORNE) //infect bystanders
		infective = TRUE

:rofl: 

Fixes cough symptom from setting virus spread to 0 (lol), fixes beard virus not working, and buffs the undead virus to both work on corpses and be worth using (transmit set to 2).

Closes #42975

Closes #42976

## Changelog
:cl:
fix: Cough, beard, and undead virus now work properly
/:cl:

### CODE REVIEW

**Review by SpaceManiac — DISMISSED** [review]

**Comment by XDTM** [comment]
Dont bundle buffs with fixes

**Comment by ATH1909** [comment]
Good PR; please merge this.

> Dont bundle buffs with fixes

In the future, yeah, don't do that, but I don't want to have the merging of this fix delayed by people demanding that the minor buff should be taken out.

File: code/datums/diseases/advance/symptoms/beard.dm [hunk 1]
```diff
@@ -27,25 +27,17 @@ BONUS
...
-				to_chat(H, "<span class='warning'>You feel tough.</span>")
-				if(!(H.facial_hair_style == "Dwarf Beard") && !(H.facial_hair_style == "Very Long Beard") && !(H.facial_hair_style == "Full Beard"))
-					H.facial_hair_style = "Full Beard"
-					H.update_hair()
-			else
-				to_chat(H, "<span class='warning'>You feel manly!</span>")
-				if(!(H.facial_hair_style == "Dwarf Beard") && !(H.facial_hair_style == "Very Long Beard"))
-					H.facial_hair_style = pick("Dwarf Beard", "Very Long Beard")
-					H.update_hair()
+		var/index = min(beard_order.Find(H.facial_hair_style)+1, A.stage-1, beard_order.len)
```
Comment (ATH1909) [inline code/datums/diseases/advance/symptoms/beard.dm]: Won't this make a very long beard (or any other beard that's on the list) SHRINK (becoming a Jensen beard) at stage 2, then eventually grow back up to being a very long beard by stage 5?

In other words, change this to being:
var/index = min(max(beard_order.Find(H.facial_hair_style)+1, A.stage-1), beard_order.len)

**Comment by pubby** [comment]
I'll move the stat buff to the other PR.

**Comment by ExcessiveUseOfCobblestone** [comment]
> In the future, yeah, don't do that, but I don't want to have the merging of this fix delayed by people demanding that the minor buff should be taken out.

Then make sure you don't append any balance changes to your fix PRs or they'll become the central point.

File: code/datums/diseases/advance/symptoms/beard.dm [hunk 2]
```diff
@@ -27,25 +27,17 @@ BONUS
...
-					H.facial_hair_style = "Full Beard"
-					H.update_hair()
-			else
-				to_chat(H, "<span class='warning'>You feel manly!</span>")
-				if(!(H.facial_hair_style == "Dwarf Beard") && !(H.facial_hair_style == "Very Long Beard"))
-					H.facial_hair_style = pick("Dwarf Beard", "Very Long Beard")
-					H.update_hair()
+		var/index = min(max(beard_order.Find(H.facial_hair_style)+1, A.stage-1), beard_order.len)
+		if(index > 0 && H.facial_hair_style != beard_order[index])
+			to_chat(H, "<span class='warning'>Your chin itches.</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/diseases/advance/symptoms/beard.dm]: you've lost some of the messages, intended?

Comment (pubby) [reply code/datums/diseases/advance/symptoms/beard.dm]: Yeah.

**Pull request merged**